src/Microsoft.Azure.Batch.SoftwareEntitlement.Client.Native/SoftwareEntitlementClient.cpp

```diff
@@ -389,6 +390,92 @@ class Curl
 
         return CURLE_OK;
     }
+
+    static void ThrowIfWinHttpError(bool error, const char* source)
+    {
+        if (!error)
+        {
+            return;
+        }
+        throw Exception("Error " + std::to_string(GetLastError()) + " in " + source);
```

Comment:
Throw `std::system_error` instead, and the Windows C++ library will automatically populate the message:

```cpp
auto errCode = GetLastError();
throw std::system_error{static_cast<int>(errCode), std::system_category{}};
```

Reply:
Updated example - you can pass it as the third argument.